theseus/src/api/settings.rs

```diff
@@ -42,3 +60,88 @@ pub async fn set(settings: Settings) -> crate::Result<()> {
     State::sync().await?;
     Ok(())
 }
+
+/// Sets the new config dir, the location of all Theseus data except for the settings.json and caches
+/// Takes control of the entire state and blocks until completion
+pub async fn set_config_dir(new_config_dir: PathBuf) -> crate::Result<()> {
```

Comment:
I think this func should have a progress bar for display in the frontend. It could be per-path